pyresample/geometry.py

```diff
@@ -1275,54 +1299,64 @@ def get_lonlats(self, nprocs=None, data_slice=None, cache=False, dtype=None):
-                nprocs = self.nprocs
-
-            # Proj.4 definition of target area projection
-            if nprocs > 1:
-                target_proj = Proj_MP(**self.proj_dict)
-            else:
-                target_proj = Proj(**self.proj_dict)
-
-            # Get coordinates of local area as ndarrays
-            target_x, target_y = self.get_proj_coords(
-                data_slice=data_slice, dtype=dtype)
-
-            # Get corresponding longitude and latitude values
-            lons, lats = target_proj(target_x, target_y, inverse=True,
-                                     nprocs=nprocs)
-            lons = np.asanyarray(lons, dtype=dtype)
-            lats = np.asanyarray(lats, dtype=dtype)
+        if self.lons is not None:
+            # Data is cache already
+            lons = self.lons
+            lats = self.lats
+            if data_slice is not None:
+                lons = lons[data_slice]
+                lats = lats[data_slice]
+            return lons, lats
+
+        # Get X/Y coordinates for the whole area
+        target_x, target_y = self.get_proj_coords(data_slice=data_slice, chunks=chunks, dtype=dtype)
+        if nprocs is None and not hasattr(target_x, 'chunks'):
+            nprocs = self.nprocs
+        if nprocs is not None and hasattr(target_x, 'chunks'):
+            # we let 'get_proj_coords' decide if dask arrays should be made
+            # but if the user provided nprocs then this doesn't make sense
+            raise ValueError("Can't specify 'nprocs' and 'chunks' at the same time")
+
+        # Proj.4 definition of target area projection
+        if hasattr(target_x, 'chunks'):
+            # we are using dask arrays, map blocks to th
+            from dask.array import map_blocks
+            res = map_blocks(invproj, target_x, target_y,
+                             chunks=(target_x.chunks[0], target_x.chunks[1], 2),
+                             new_axis=[2], proj_dict=self.proj_dict)
+            return res[:, :, 0], res[:, :, 1]
+
+        if nprocs > 1:
+            target_proj = Proj_MP(**self.proj_dict)
+        else:
+            target_proj = Proj(**self.proj_dict)
 
-            if cache and data_slice is None:
-                # Cache the result if requested
-                self.lons = lons
-                self.lats = lats
+        # Get corresponding longitude and latitude values
+        lons, lats = target_proj(target_x, target_y, inverse=True, nprocs=nprocs)
+        lons = np.asanyarray(lons, dtype=dtype)
+        lats = np.asanyarray(lats, dtype=dtype)
 
-            # Free memory
-            del(target_x)
-            del(target_y)
-        else:
-            # Data is cached
-            if data_slice is None:
-                # Full slice
-                lons = self.lons
-                lats = self.lats
-            else:
-                lons = self.lons[data_slice]
-                lats = self.lats[data_slice]
+        if cache and data_slice is None:
```

Comment:
Should we deprecate caching ?

Reply:
Yes, that's why there is a PendingDeprecationWarning at the top of the method.

Reply:
Good